test/integration/targets/aws_secret/tasks/main.yaml

```diff
@@ -0,0 +1,273 @@
+---
+- block:
+  - name: set connection information for all tasks
+    set_fact:
+      aws_connection_info: &aws_connection_info
+        aws_access_key: "{{ aws_access_key }}"
+        aws_secret_key: "{{ aws_secret_key }}"
+        region: "{{ aws_region }}"
+        security_token: "{{ security_token }}"
+    no_log: true
+
+  - name: retrieve caller facts
+    aws_caller_facts:
+      <<: *aws_connection_info
+    register: test_caller_facts
+
+  - name: ensure IAM role exists
+    iam_role:
+      <<: *aws_connection_info
+      name: "{{ resource_prefix }}-test-secrets-manager-role"
```

Comment:
Please prefix the role name with "ansible-test-secrets-" with the resource_prefix following so it's easier to add resource restrictions to the tests.

Reply:
(or a similar prefix)